YelpReview/resultViewController.swift

```diff
@@ -0,0 +1,235 @@
+        var randomLabel = self.loadingLabels[randomIndex]
+        randomIndex = Int(arc4random_uniform(UInt32(self.lw.count)))
+        var randomWord = self.lw[randomIndex]
+        let randomNumber = String(arc4random_uniform(UInt32(20)))
+        var result =
+        randomLabel.stringByReplacingOccurrencesOfString(":N:", withString: randomNumber, options: NSStringCompareOptions.LiteralSearch, range: nil).stringByReplacingOccurrencesOfString(":W:", withString: randomWord, options: NSStringCompareOptions.LiteralSearch, range: nil)
+        return result
+    }
+    
+    @IBOutlet var floatRatingView: FloatRatingView!
+    @IBOutlet weak var businessType: UILabel!
+    //@IBOutlet weak var stars: UILabel!
+    @IBOutlet weak var resultText: UITextView!
+    @IBOutlet weak var generateButton: UIButton!
+    @IBOutlet weak var resultContainer: UIView!
+    var approaches = 3
+    var back: UIImageView!
+    var front: UIImageView!
+    var result: UITextView!
+    var loading = false
+    var shareText = ""
+    var shareUrl = ""
+    @IBOutlet weak var loadingView: UIView!
+    
+    @IBOutlet weak var loadingTextView: UITextView!
+    
+    func startLoading() {
+        self.loading = true
+        self.generateButton.hidden = true
+        self.shareView.hidden = true
+        self.loadingView.hidden = false
+        self.loadingTextView.text = getLoadingLabel()
+        self.loadingTextView.font = UIFont(name: "Helvetica", size: 16)
+        self.MonkeyFace.alpha = 1
+
+
+    }
+    
+    func stopLoading() {
+        if self.regenerations == self.approaches {
+            self.generateButton.setTitle("New review", forState: UIControlState.Normal)
+            self.generateButton.backgroundColor = self.UIColorFromRGB(0x81451D)
+            self.generateButton.tintColor = self.UIColorFromRGB(0xFFFFFF)
+
+        }
+        self.loadingView.hidden = true
+        self.loading = false
+        self.generateButton.hidden = false
+        self.shareView.hidden = false
+        UIView.animateWithDuration(0.5, animations: {
+            self.resultText.alpha = 1
+            self.MonkeyFace.alpha = 0
+        })
+    }
+
+    // Main API call
+    func getRating(stars: Int, business: String, completion: (rating: String) -> Void) {
+        var url = NSURL(string: "http://www.reviewmonkeyapp.com/review/\(stars)/\(business).json")
+        let task = NSURLSession.sharedSession().dataTaskWithURL(url!, completionHandler: { (data: NSData!, response: NSURLResponse!, error: NSError!) -> Void in
+            dispatch_async(dispatch_get_main_queue(), { () -> Void in
+                var error: NSError?;
+                let options = NSJSONReadingOptions(0)
+                let jsonObject: AnyObject? = NSJSONSerialization.JSONObjectWithData(data!, options: options, error: &error)
+                
+                // Call closure
+                if let decodedObject = jsonObject as? NSDictionary {
```

Comment:
better to make a review model that takes the json object in the constructor. Also you should use decodedObject["review"] as? String